esp-wifi/src/timer/riscv.rs

```diff
@@ -34,7 +32,19 @@ pub fn setup_timer(mut alarm0: TimeBase) -> Result<(), esp_hal::timer::Error> {
     alarm0.start(TIMESLICE_FREQUENCY.into_duration())?;
     critical_section::with(|cs| {
         alarm0.enable_interrupt(true);
-        ALARM0.borrow_ref_mut(cs).replace(alarm0);
+        TIMER.borrow_ref_mut(cs).replace(alarm0);
+    });
+
+    Ok(())
+}
+
+pub fn disable_timer() -> Result<(), esp_hal::timer::Error> {
+    // make sure the scheduling won't start before everything is setup
```

Comment:
at least the comment is misleading - also probably unnecessary to globally disable (and later enable) interrupts